Modules/Quotas/src/Quotas.Infrastructure/Persistence/Database/EntityConfigurations/TierQuotaEntityTypeConfiguration.cs

```diff
@@ -0,0 +1,17 @@
+﻿using Backbone.Modules.Quotas.Domain.Aggregates.Identities;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+
+namespace Backbone.Modules.Quotas.Infrastructure.Persistence.Database.EntityConfigurations;
+
+public class TierQuotaEntityTypeConfiguration : IEntityTypeConfiguration<TierQuota>
+{
+    public void Configure(EntityTypeBuilder<TierQuota> builder)
+    {
+        builder.UseTpcMappingStrategy();
+        builder.HasKey(x => x.Id);
+        builder.Property(x => x.ApplyTo);
+        builder.Property(x => x.IsExhaustedUntil);
```

Comment:
That's the same as what EF Core does by convention. So you can remove all of this. Or doesn't this work because of the inheritance?

Reply:
Like stated in the TierQuotaDefinitionEntityTypeConfiguration, the lack of an HasKey call results in:
`The entity type 'TierQuota' requires a primary key to be defined.`

In this case, I can remove the ApplyTo Property implicit call, however the call for IsExhaustedUntil is required, otherwise it's removed from the migration:

```
protected override void Up(MigrationBuilder migrationBuilder)
{
    migrationBuilder.DropColumn(
        name: "IsExhaustedUntil",
        table: "TierQuotas");
}
```